app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt

```diff
@@ -22,38 +23,45 @@ import android.content.SharedPreferences
 import android.text.Editable
 import android.text.TextWatcher
 import android.view.View
+import android.view.View.VISIBLE
 import android.view.inputmethod.InputMethodManager
 import android.widget.CheckBox
 import android.widget.EditText
 import androidx.constraintlayout.widget.ConstraintLayout
+import com.google.android.material.textfield.TextInputLayout
 import com.yacgroup.yacguide.R
 
 class SearchBarHandler(searchBarLayout: ConstraintLayout,
                        searchHintResource: Int,
-                       checkBoxTitle: String,
-                       checkBoxDefaultValue: Boolean,
-                       private val _settings: SharedPreferences,
-                       private val _initCallback: (Boolean) -> Unit,
-                       private val _updateCallback: (String, Boolean) -> Unit) {
+                       checkBoxTitle: String = "",
+                       checkBoxDefaultValue: Boolean = false,
+                       private val _settings: SharedPreferences? = null,
+                       private val _initCallback: ((Boolean) -> Unit)? = null,
```

Comment:
Wouldn't it be safer (and more convenient) to default `_initCallback` and `_updateCallback` to empty lambdas?
--> `_initCallback: (Boolean) -> Unit = {}`